Gateway/Command/AbstractCommand.php

```diff
@@ -0,0 +1,342 @@
+     * @param ClientRequestService $requestService
+     * @param MageQuoteRepository $mageQuoteRepo
+     * @param MageOrderRepository $mageOrderRepo
+     * @param Logger $logger
+     * @param array $data
+     */
+    public function __construct(
+        PaymentOrderRepository $paymentOrderRepo,
+        PaymentQuoteRepository $paymentQuoteRepo,
+        ClientRequestService $requestService,
+        MageQuoteRepository $mageQuoteRepo,
+        MageOrderRepository $mageOrderRepo,
+        Logger $logger,
+        array $data = []
+    ) {
+        parent::__construct($data);
+        $this->paymentOrderRepo = $paymentOrderRepo;
+        $this->paymentQuoteRepo = $paymentQuoteRepo;
+        $this->requestService = $requestService;
+        $this->mageQuoteRepo = $mageQuoteRepo;
+        $this->mageOrderRepo = $mageOrderRepo;
+        $this->logger = $logger;
+
+        $this->cmdTransActionMap = [
+            self::GATEWAY_COMMAND_CAPTURE => self::TRANSACTION_ACTION_CAPTURE,
+            self::GATEWAY_COMMAND_CANCEL => self::TRANSACTION_ACTION_CANCEL,
+            self::GATEWAY_COMMAND_REFUND => self::TRANSACTION_ACTION_REFUND
+        ];
+    }
+
+    /**
+     * AbstractCommand command
+     *
+     * @param array $commandSubject
+     *
+     * @return null|Command\ResultInterface
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    abstract public function execute(array $commandSubject);
+
+    /**
+     * Get client request service class
+     *
+     * @param string $service
+     * @param string $operation
+     * @param AbstractDataTransferObject|null $dataTransferObject
+     * @return RequestInterface|string
+     * @throws ServiceException
+     */
+    protected function getRequestService($service, $operation, $dataTransferObject = null)
+    {
+        return $this->requestService->init($service, $operation, $dataTransferObject);
+    }
+
+    /**
+     * Generates a random string
+     *
+     * @param $length
+     * @return bool|string
+     */
+    protected function generateRandomString($length)
+    {
+        return substr(str_shuffle(hash('sha256', time())), 0, $length);
+    }
+
+    /**
+     * @param string $command
+     * @param ResponseResourceInterface $responseResource
+     * @param OrderInterface $mageOrder
+     * @param string $paymentId
+     * @throws SwedbankPayException
+     */
+    protected function checkResponseResource($command, $responseResource, $mageOrder, $paymentId)
+    {
+        if ($responseResource instanceof ResponseResourceInterface) {
+            return;
+        }
+
+        $this->logger->error(
+            sprintf(
+                "Failed to %s order %s with SwedbankPay payment id %s, response resource:\n%s",
+                $command,
+                $mageOrder->getEntityId(),
+                $paymentId,
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                print_r($responseResource, true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function print_r() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)